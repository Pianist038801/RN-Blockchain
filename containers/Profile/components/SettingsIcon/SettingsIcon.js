import React from 'react'
import PropTypes from 'prop-types'
import { TouchableOpacity, Image, View, Text } from 'react-native'
import { Actions } from 'react-native-router-flux'
import style, { height } from '../../../../style'

const wrapper = {
  height: 54,
  width: 54,
  marginTop: -20,
  marginLeft: -15,
  paddingTop: 12,
  paddingLeft: 20
}
const badgeContainer = {
  ...style.badgeContainer,
  width: height * 0.034,
  height: height * 0.034,
  borderRadius: height * 0.017,
  left: 15,
  top: 7
}

const SettingsIcon = (props) => {
  const { profile } = props
  const renderNotification = profile && profile.first_name &&
    !['complete', 'needs_review'].includes(profile.biometrics_status)

  return (
    <TouchableOpacity
      onPress={Actions.settings}
      style={wrapper}
    >
      <Image
        source={require('../../../../assets/img/settings.png')}
        style={style.settingsBtn}
      />
      {
        renderNotification &&
        <View style={badgeContainer}>
          <Text style={style.badgeText}>!</Text>
        </View>
      }
    </TouchableOpacity>
  )
}

SettingsIcon.propTypes = {
  profile: PropTypes.shape({
    first_name: PropTypes.string,
    biometrics_status: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.object
    ])
  })
}

export default SettingsIcon
