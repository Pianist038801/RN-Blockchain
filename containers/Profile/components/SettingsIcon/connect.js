import { connect } from 'react-redux'
import Component from './SettingsIcon.js'

export const mapState = (state) => ({
  profile: state.profile
})

export default connect(mapState, {})(Component)
