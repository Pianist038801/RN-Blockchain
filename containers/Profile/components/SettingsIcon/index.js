export { default, mapState } from './connect'
