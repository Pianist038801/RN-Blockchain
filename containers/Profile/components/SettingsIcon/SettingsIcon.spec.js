import React from 'react'
import renderer from 'react-test-renderer'
import SettingsIcon from './SettingsIcon'

describe('containers/Profile/components/SettingsIcon', () => {
  describe('when user is verified', () => {
    it('renders correctly', () => {
      const props = {
        profile: {
          first_name: 'John',
          biometrics_status: 'complete'
        }
      }

      const tree = renderer.create(
        <SettingsIcon {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when user is in review', () => {
    it('renders correctly', () => {
      const props = {
        profile: {
          first_name: 'John',
          biometrics_status: 'needs_review'
        }
      }

      const tree = renderer.create(
        <SettingsIcon {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when state is loading', () => {
    it('renders correctly', () => {
      const props = {
        profile: {
          first_name: null,
          biometrics_status: null
        }
      }

      const tree = renderer.create(
        <SettingsIcon {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when user is not verified', () => {
    it('renders correctly', () => {
      const props = {
        profile: {
          first_name: 'John',
          biometrics_status: null
        }
      }

      const tree = renderer.create(
        <SettingsIcon {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when verification has failed', () => {
    it('renders correctly', () => {
      const props = {
        profile: {
          first_name: 'John',
          biometrics_status: 'failed'
        }
      }

      const tree = renderer.create(
        <SettingsIcon {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
