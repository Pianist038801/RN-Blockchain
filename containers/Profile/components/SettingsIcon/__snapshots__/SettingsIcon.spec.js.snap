// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Profile/components/SettingsIcon when state is loading renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 54,
      "marginLeft": -15,
      "marginTop": -20,
      "opacity": 1,
      "paddingLeft": 20,
      "paddingTop": 12,
      "width": 54,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Image
    source={1}
    style={
      Object {
        "height": 35,
        "resizeMode": "contain",
        "width": 35,
      }
    }
  />
</View>
`;

exports[`containers/Profile/components/SettingsIcon when user is in review renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 54,
      "marginLeft": -15,
      "marginTop": -20,
      "opacity": 1,
      "paddingLeft": 20,
      "paddingTop": 12,
      "width": 54,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Image
    source={1}
    style={
      Object {
        "height": 35,
        "resizeMode": "contain",
        "width": 35,
      }
    }
  />
</View>
`;

exports[`containers/Profile/components/SettingsIcon when user is not verified renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 54,
      "marginLeft": -15,
      "marginTop": -20,
      "opacity": 1,
      "paddingLeft": 20,
      "paddingTop": 12,
      "width": 54,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Image
    source={1}
    style={
      Object {
        "height": 35,
        "resizeMode": "contain",
        "width": 35,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "red",
        "borderRadius": 22.678,
        "height": 45.356,
        "justifyContent": "center",
        "left": 15,
        "position": "absolute",
        "right": -10,
        "top": 7,
        "width": 45.356,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Ubuntu",
          "fontSize": 26.68,
        }
      }
    >
      !
    </Text>
  </View>
</View>
`;

exports[`containers/Profile/components/SettingsIcon when user is verified renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 54,
      "marginLeft": -15,
      "marginTop": -20,
      "opacity": 1,
      "paddingLeft": 20,
      "paddingTop": 12,
      "width": 54,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Image
    source={1}
    style={
      Object {
        "height": 35,
        "resizeMode": "contain",
        "width": 35,
      }
    }
  />
</View>
`;

exports[`containers/Profile/components/SettingsIcon when verification has failed renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 54,
      "marginLeft": -15,
      "marginTop": -20,
      "opacity": 1,
      "paddingLeft": 20,
      "paddingTop": 12,
      "width": 54,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Image
    source={1}
    style={
      Object {
        "height": 35,
        "resizeMode": "contain",
        "width": 35,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "red",
        "borderRadius": 22.678,
        "height": 45.356,
        "justifyContent": "center",
        "left": 15,
        "position": "absolute",
        "right": -10,
        "top": 7,
        "width": 45.356,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Ubuntu",
          "fontSize": 26.68,
        }
      }
    >
      !
    </Text>
  </View>
</View>
`;
